pkg/reconciler/buildrun/buildrun.go

```diff
@@ -147,16 +147,25 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			if err != nil {
 				return reconcile.Result{}, err
 			}
-
 			ctxlog.Info(ctx, "creating TaskRun from BuildRun", namespace, request.Namespace, name, generatedTaskRun.GenerateName, "BuildRun", buildRun.Name)
 			if err = r.client.Create(ctx, generatedTaskRun); err != nil {
 				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
+				serviceAccountName := &generatedTaskRun.Spec.ServiceAccountName
+				buildRun.Status.ServiceAccountName = serviceAccountName
+				if err = r.client.Status().Update(ctx, buildRun); err != nil {
+					// we ignore the error here to prevent another reconciliation that would create another TaskRun,
+					// the LatestTaskRunRef field will also be set in the reconciliation from a TaskRun
+					// risk is that when the controller is now restarted before the field is set, another TaskRun will be created
+					ctxlog.Error(ctx, err, "Failed to update BuildRun status is ignored", namespace, request.Namespace, name, request.Name)
```

Comment:
Nit: The error message is a bit hard to read, maybe we reverse the order and start the sentence with the fact that we **ignore** the following described situation.
```suggestion
					ctxlog.Error(ctx, err, "ignoring that BuildRun status update has failed for setting the service account name", namespace, request.Namespace, name, request.Name)
```